packages/lab/src/accordion/Accordion.tsx

```diff
@@ -77,7 +77,9 @@ export const Accordion = forwardRef<HTMLDivElement, AccordionProps>(
         });
         if (isExpanded) {
           setExpandedSectionIds((oldExpandedSectionIds) => {
-            const newExpandedSectionIds = [...oldExpandedSectionIds];
+            const newExpandedSectionIds = ([] as string[]).concat(
```

Comment:
This component seems to have some issues. Are these changes relevant to the PR? Maybe they should be split out?

Reply:
It's relevant. This is only an issue because the move to React causes types to be more "correct" so it was only flagged here. Happy to move it to another PR though

Reply:
Think I might be missing something here. If I check out the branch `oldExpandedSectionIds` seems to be types as `string` and TS seems to be happy with the original spread.

Is it possible for `expandedSectionIds` to be `undefined`? If so should the `useControlled<string[]>` above be `useControlled<string[] | undefined>`?

Or maybe `AccordionProps` type should be changed so that either `expandedSectionIds` or `defaultExpandedSectionIds` has to be defined?

Reply:
Due to the changes in useControlled undefined is possible. So there doesn't need to be an explicit undefined above.